### PR TITLE
Automatic generation of keys for root tables

### PR DESCRIPTION
**Pull request by jakobbraun**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.GlobalSecondaryIndexDescription;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.LocalSecondaryIndexDescription;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+
+/**
+ * This class builds {@link DynamodbTableMetadata} by fetching the required information using a {@code describeTable}
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java:15]:
```suggestion
 * This class builds {@link DynamodbTableMetadataFactory} by fetching the required information using a {@code describeTable}
```

Comment (morazow) [reply src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java:15]: Or `DynamodbTableMetadataFactoryImplementation`.

File: src/main/java/com/exasol/adapter/dynamodb/mapping/RootTableMappingReader.java [hunk 2]
```diff
@@ -12,30 +13,40 @@
...
     @Override
-    protected GlobalKey generateGlobalKey() {
-        throw new UnsupportedOperationException("fetching keys from remote database is not yet supported"); // TODO
+    protected GlobalKey generateGlobalKey(final List<ColumnMapping> availableColumns) {
+        try {
+            return new GlobalKey(Collections.emptyList(),
+                    this.tableKeyFetcher.fetchKeyForTable(this.remoteTableName, availableColumns));
+        } catch (final TableKeyFetcher.NoKeyFoundException e) {
+            throw new ExasolDocumentMappingLanguageException("Could not infer keys for table " + this.exasolTableName
+                    + ". Please mark define a unique key by setting key='global' for one or more columns.");
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/dynamodb/mapping/RootTableMappingReader.java:48]: Remove one of the verbs, `mark` and `define` both are verb.

File: src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java (hunk 1 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java:15]: no, this is the factory for `DynamodbTableMetadata`. `DynamodbTableMetadataFactory` is  the interface for the factory and `DynamodbTableMetadataFactoryImplementation` the implementation. Should I change the names somehow?

Comment (morazow) [reply src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java:15]: Ahh okay. I get it now. It builds `DynamodbTableMetadata`. 
The all is okay from my side. But maybe you can remove the `Implementation` suffix and add `Base`, e.g, `BaseDynamodbTableFactory`. But it is up to you, please feel free to choose any naming.

**Pull request merged**